pyfr/plugins/tavg.py

```diff
@@ -220,6 +226,46 @@ def _acc_avg_var(self, intg, currex):
             if self.tstart_acc != self.prevt:
                 v +=  (Wmp1mpn / Wp * (a - W1mpn * ppc)**2)
 
+    @staticmethod
+    def merge_cli(args):
+        a_out = h5py.File(args.output, 'w')
+        start_time = np.empty((0, 1))
+        end_time = np.empty((0, 1))
+
+        for i, filename in enumerate(args.solns):
+            a_in = h5py.File(filename, 'r')
+
+            stat = Inifile(a_in['stats'][()].decode('UTF-8'))
+            end_time = np.append(end_time, stat.getfloat('tavg', 'tend'))
+            start_time = np.append(start_time, stat.getfloat('tavg', 'tstart'))
+            dt = end_time[-1] - start_time[-1]
+
+            if i == 0:
+                uuid = a_in['mesh_uuid'][()]
+                stat_base = stat
+                data_keys = [k for k in a_in.keys() if k.startswith('tavg_')]
```

Comment:
Some extra sanity checks are probably worthwhile here (disallow overlapping time ranges unless a flag is passed, check the field names are all the same).